src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -55,16 +55,67 @@ class Gemini(session: SparkSession, log: Slf4jLogger, keyspace: String = Gemini.
       .getCommits
       .getTreeEntries
       .getBlobs
+      .filter('is_binary === false)
+
+  def sparkExtractUast(files: DataFrame): DataFrame = {
+    // TODO(max): get languages from bblfsh directly as soon as
+    // https://github.com/bblfsh/client-scala/issues/68 resolved
+    val langs = Array("Java", "Python", "Go", "JavaScript", "TypeScript", "Ruby", "Bash", "Php")
+
+    files
+      .dropDuplicates("blob_id")
+      .classifyLanguages
+      .filter('lang.isin(langs: _*))
+      .extractUASTs
+      .select("repository_id", "path", "blob_id", "uast")
+      .filter(_.getAs[Seq[Array[Byte]]]("uast").nonEmpty)
+      .withColumn("document", expr("CONCAT(repository_id, '//', path, '@', blob_id)"))
+      .select("document", "uast")
+  }
+
+  def sparkFeatures(uasts: DataFrame): RDD[SparkFeature] = {
+    var feConfig = SparkFEClient.getConfig(session)
+    val rows = uasts.flatMap { row =>
```

Comment:
As apache spark iteration is followed by collection iteration here, adding a suffix to var names would communicate it clearer i.e `uastsDf` and `rowsDf` would contrast to `uastArr` and `features`.